xonsh/tools.py

```diff
@@ -1171,7 +1171,7 @@ def backup_file(fname):
     import shutil
     from datetime import datetime
     base, ext = os.path.splitext(fname)
-    newfname = base + '.' + datetime.now().isoformat() + ext
+    newfname = '%s.%s%s' % (base, datetime.now().isoformat().replace(':', '-'), ext)
```

Comment:
I wanted to avoid using `strftime` because according to `isoformat`'s docs it implements some special behavior depending on `datetime`'s global state:

> If utcoffset() does not return None, a 6-character string is appended, giving the UTC offset in (signed) hours and minutes: YYYY-MM-DDTHH:MM:SS.mmmmmm+HH:MM or, if microsecond is 0 YYYY-MM-DDTHH:MM:SS+HH:MM

But you are right, preserving that behaviour is irrelevant for this. I'll make the suggested fixes.